engine/lite/k8sDeploymentManager/src/main/scala/pl/touk/nussknacker/k8s/manager/ingress/IngressPreparer.scala

```diff
@@ -0,0 +1,60 @@
+package pl.touk.nussknacker.k8s.manager.ingress
+
+import com.typesafe.config.{Config, ConfigFactory, ConfigRenderOptions}
+import monocle.macros.GenLens
+import pl.touk.nussknacker.engine.api.{LiteStreamMetaData, ProcessVersion, RequestResponseMetaData, TypeSpecificData}
+import pl.touk.nussknacker.k8s.manager.K8sDeploymentManager.{labelsForScenario, objectNamePrefixedWithNussknackerInstanceNameWithoutSanitization}
+import pl.touk.nussknacker.k8s.manager.K8sUtils.sanitizeObjectName
+import pl.touk.nussknacker.k8s.manager.RequestResponseSlugUtils
+import pl.touk.nussknacker.k8s.manager.ingress.IngressPreparer.rewriteAnnotation
+import play.api.libs.json.Json
+import skuber.networking.v1.Ingress
+import monocle.std.option.some
+
+case class IngressConfig(host: String, config: Config = ConfigFactory.empty())
+
+class IngressPreparer(config: IngressConfig, nuInstanceName: Option[String]) {
+
+  def prepare(processVersion: ProcessVersion, typeSpecificData: TypeSpecificData, serviceName: String, servicePort: Int): Option[Ingress] =
+    typeSpecificData match {
+      case _: LiteStreamMetaData => None
+      case rrMetaData: RequestResponseMetaData => Some(prepareRequestResponseIngress(processVersion, rrMetaData, serviceName, servicePort))
+      case other => throw new IllegalArgumentException("Not supported scenario meta data type: " + other)
+    }
+
+  private def prepareRequestResponseIngress(processVersion: ProcessVersion, rrMetaData: RequestResponseMetaData, serviceName: String, servicePort: Int): Ingress = {
+    val objectName = IngressPreparer.name(nuInstanceName, s"scenario-${processVersion.processName.value}")
+    val labels = labelsForScenario(processVersion, nuInstanceName)
+    val slug = RequestResponseSlugUtils.determineSlug(processVersion.processName, rrMetaData, nuInstanceName)
+
+    //we use 'OptionOptics some' here and do not worry about withDefault because _.spec is provided in defaultMinimalIngress
+    val ingressSpecLens = GenLens[Ingress](_.spec) composePrism some
+
+    val lens = GenLens[Ingress](_.metadata.name).set(objectName) andThen
+      GenLens[Ingress](_.metadata.labels).modify(_ ++ labels) andThen
+      GenLens[Ingress](_.metadata.annotations).modify(_ ++ rewriteAnnotation) andThen
+      (ingressSpecLens composeLens GenLens[Ingress.Spec](_.rules)).modify(_ ++ List(
+        Ingress.Rule(Some(config.host), Ingress.HttpRule(paths = List(Ingress.Path(
+          path = s"/$slug" + "(/|$)(.*)",
```

Comment:
I assume that changes in openapi definition will be done in other PR? Do we have a task for that? Maybe at least add somewhere TODO that open api will have bad url for ingress for now

Reply:
Yes, we have task for that. I also added a comment here.